### PR TITLE
Basic pcapng writing support

### PR DESCRIPTION
**Pull request by guedou**
This PR adds a basic pcapng writing support to Scapy.

### CODE REVIEW

**Comment by guedou** [comment]
This PR is ready to be reviewed. It works fine and does not break tests locally, however I am not satisfied with the current design. Comments are welcomed!

@gpotter2 do you know how I can solve these mypy issues?
```
scapy/utils.py:1735: error: "GenericRawPcapWriter" has no attribute "_write_header"; maybe "write_header"?
scapy/utils.py:1764: error: "GenericRawPcapWriter" has no attribute "_write_packet"; maybe "write_packet"?
scapy/utils.py:2112: error: "GenericPcapWriter" has no attribute "_write_header"; maybe "write_header"?
scapy/utils.py:2122: error: Item "bytes" of "Union[bytes, Packet]" has no attribute "time"   
scapy/utils.py:2173: error: "GenericPcapWriter" has no attribute "_write_packet"; maybe "write_packet"?
```

File: scapy/utils.py [hunk 1]
```diff
@@ -1701,7 +1719,85 @@ def recv(self, size=MTU):
         return self.read_packet()
 
 
-class RawPcapWriter(object):
+class GenericRawPcapWriter(object):
```
Comment (gpotter2) [inline scapy/utils.py:1722]: You could add an unimplemented `_write_header` function to this class (raising NotImplementedError)

File: scapy/utils.py [hunk 2]
```diff
@@ -1918,13 +2106,30 @@ def write_header(self, pkt):
...
                 self.linktype = DLT_EN10MB
         self._write_header(pkt)
 
+    def _get_time(self,
+                  packet,  # type: Union[bytes, Packet]
+                  sec,  # type: Optional[float]
+                  usec  # type: Optional[int]
+                  ):
+        # type: (...) -> Tuple[float, int]
+        if hasattr(packet, "time"):
```
Comment (gpotter2) [inline scapy/utils.py:2120]: couldn't this be replaced with `isinstance(packet Packet)` ? or not isinstance bytes

**Comment by gpotter2** [comment]
Nice PR ! Few suggestions for mypy

File: scapy/utils.py (hunk 2 above)
Comment (guedou) [reply scapy/utils.py:2120]: That makes sense. The typing will be cleaner.

**Comment by guedou** [comment]
Checks are now green!

I reverted instance checks as they introduces a circular dependency.

The regression test change looks OK as the linktype is now set to a default value earlier than before.

**Review by p-l- — APPROVED** [review]

**Review by gpotter2 — APPROVED** [review]

**Pull request merged**